ui/desktop/app/controllers/scopes/scope/projects/targets/index.js

```diff
@@ -0,0 +1,103 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+import Controller from '@ember/controller';
+import { inject as service } from '@ember/service';
+import { action } from '@ember/object';
+import { loading } from 'ember-loading';
+
+export default class ScopesScopeProjectsTargetsIndexController extends Controller {
+  // =services
+
+  @service confirm;
+  @service ipc;
+  @service router;
+  @service session;
+  @service store;
+
+  // =methods
+
+  /**
+   * Quick connect method used to call main connect method and handle
+   * connection errors unique to this route
+   * @param {TargetModel} target
+   */
+  @action
+  async quickConnect(target) {
+    try {
+      await this.connect(target);
+    } catch (error) {
+      this.confirm
+        .confirm(error.message, { isConnectError: true })
+        // Retry
+        .then(() => this.quickConnect(target));
+    }
+  }
+
+  /**
+   * Establish a session to current target.
+   * @param {TargetModel} target
+   * @param {HostModel} host
+   */
+  @action
+  @loading
```

Comment:
Should the `loading` decorator be on this or the connect functions that use these? Or can we decorate them all with `loading`?

Reply:
That is a good point. I will document this for when we refactor the error messages part of this work.